### ISSUE TITLE
Softlocked character on Giant creature throw

### ISSUE THREAD

**Issue by Nestastnikos** [FishingCactus/NanotaleEarlyAccessBugs#64]
**Describe the bug**
After the Rosalind is in typing mode and the Giant creature will push the Rosalind, the typing mode is broken and the Rosalind is softlocked.

**To Reproduce**
1. Precondition: Engaged in the fight with the giant creature.
2. Let the creature throw the Rosalind.

**Actual behavior**
The typing mode is interrupted and the Rosalind's movements are softlocked.

**Expected behavior**
The typing mode can be interrupted, but the Rosalind should at least be able to move.

**Screenshots**
![20191101184028_1](https://user-images.githubusercontent.com/23109654/68046924-65f46c00-fcdd-11e9-9f89-73c866279023.jpg)

**Additional context**
The typing mode can be reentered again, only movements are locked. Not after every throw the softlock is triggered. I saw there is a bunch of ArgumentOutOfRangeExceptions in the logs after this bug occurs.

**Comment by Ruvah** [comment]
Hello !

It is intentional that your moving and typing is temporarely locked after being hit by this enemy as it stuns you.(We are missing some visuals to indicate that atm)

However the lock remaining longer than a second or two is not intentional. 

Further investigation will follow on how it is able to lock you permanently

**Comment by Nestastnikos** [comment]
@Ruvah Hi, well I see that I did not specify it precisely. After you are thrown by the enemy, then the softlock was permanent while you were in game. I am not quite sure but if I remember correctly the Rosalind is softlocked even after she dies (she still cannot move). The only way how to get out of this was to relaunch the game.

I am just following up on this because I see that it was closed but I don't think that the permanent lock in game is intentional here.

EDIT: Yeah it seems that it is resolved. I cannot reproduce it anymore.
